File: 02/part1.php
<?php
if (count($argv) < 2) {
	echo "Please provide data file as argument";
	exit(1);
}
$data = array_filter(explode("\n", file_get_contents($argv[1])));
$colour_max = ["red" => 12, "green" => 13, "blue" => 14];

$total = 0;
foreach($data as $line) {
	[$p1, $rest] = explode(":", $line);
	[$game, $gameNumber] = explode(" ", $p1);
	$sets = explode(";", $rest);
	foreach($sets as $set) {
		$items = explode(",", $set);
		foreach($items as $item) {
			[$space, $num, $colour] = explode(" ", $item);
			if ($num > $colour_max[$colour]) {
				continue 3;
			}
		}
	}
	$total = $total + $gameNumber;
}
var_dump($total);

File: 02/part1.py
import re
import sys

if len(sys.argv) < 2:
    print("please specify input file")
    sys.exit(1)

f = open(sys.argv[1], 'r')
data = f.read()
f.close()

colour_max = {
    "red": 12,
    "green": 13,
    "blue": 14,
}
colours = list(colour_max.keys())

games = {}

sumOfValids = 0
for line in data.splitlines():
    res = re.findall('^Game ([0-9]+):', line)
    if len(res) < 1:
        raise Exception("Cannot determine game number")
    gameNumber = res[0]
    line = line[len("Game " + gameNumber) + 2:]

    isLineValid = True
    for set in line.split(";"):
        isSetValid = True
        for colour in colours:
            res = re.findall("([0-9]+) " + colour, set)
            if len(res) > 0:
                if int(res[0]) > colour_max[colour]:
                    isSetValid = False
        if isSetValid is False:
            isLineValid = False

    if isLineValid:
        sumOfValids += int(gameNumber)
    print("num " + gameNumber + ": " + str(isLineValid))

print(sumOfValids)
